src/services/milestones/checkMilestoneName.js

```diff
@@ -0,0 +1,35 @@
+const errors = require('@feathersjs/errors');
+
+/**
+ * This function checks if milestones name is unique in the campaign scope
+ * */
+const checkIfMilestoneNameIsUnique = () => async context => {
+  const { data, app } = context;
+  if (!data.title) {
+    return context;
+  }
+  const title = data.title.trim();
+  const milestoneService = app.service('milestones');
+  if (!data.campaignId) {
+    const milestone = await milestoneService.get(context.id);
+    if (milestone) {
+      data.campaignId = milestone.campaignId;
+    }
+  }
+  const milestoneWithSameName = await milestoneService.find({
+    query: {
+      campaignId: data.campaignId,
+      title: new RegExp(`\\s*${title.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}\\s*`),
```

Comment:
I am not sure whether it's a good idea to match it case insensitive. @mdehghani what do you think?

Reply:
It's not matching case insensitive. It tries to trim the title (remove whitespaces from the beginning and the end of the title).
But it seems there is a problem in the regex. I will change it to a simpler regex.